xml2rfc_compat/serializers/reference.py

```diff
@@ -106,19 +107,30 @@ def create_reference(item: BibliographicItem) -> _Element:
             func(docid)
             for func in DOCID_SERIES_EXTRACTORS
         ])
+
+    is_doi = False
     for series_info in list(dict.fromkeys(series)):
         if series_info is not None:
             ref.append(E.seriesInfo(
                 name=series_info[0],
                 value=series_info[1],
             ))
+            if series_info[0] == "DOI":
```

Comment:
Note that this will flag RFCs which has a `DOI` in `seriesInfo` as `True`.
But RFC targets don't have doi.org URLs.
So the test for hostname in line: 130 becomes very important.
Might be worth adding a comment in line 130 for the future?

Reply:
This is correct. I am wondering if perhaps a more structured approach would be to systematically replace any http://dx.doi.org with https://doi.org and remove the rest of the checks? @kesara what's your opinion?

Reply:
So this should only happen for `bibxml-doi` dataset.
And these citations are created on-demand correct?
If so, this should happen when the citation gets created.

Reply:
Thanks for the clarification @kesara . Apologies for not understanding the problem earlier. 

I moved the URL manipulation to the DoiAdapter. I am now testing, will send another update shortly.